pathfinder/tests/test_blockchain_integration.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+
+"""
+The test in this module uses the mocked raiden client to create blockchain events and
+processes them. Additionally, it mocks the transport layer directly. It tests the
+interaction of many moving parts - yet, it is currently really slow.
+Therefore, usually mocked_integration should be used.
+"""
+
+import gevent
+from raiden_contracts.contract_manager import ContractManager
+from pathfinder.pathfinding_service import PathfindingService
+from pathfinder.tests.fixtures import Mock
+from raiden_libs.blockchain import BlockchainListener
+
+
+def test_pfs_with_mocked_client(
+    generate_raiden_clients,
+    ethereum_tester,
+    contracts_manager: ContractManager,
+    blockchain_listener: BlockchainListener
+
+):
+    """ instantiates a pathfinding service with mocked transport,
+    listening and porcessing blockchain events created by mocked Raiden clients"""
+    clients = generate_raiden_clients(7)
+    network_address = clients[0].contract.address
+
+    pathfinding_service = PathfindingService(
+        contracts_manager,
+        transport=Mock(),
+        token_network_listener=blockchain_listener,
+        follow_networks=[network_address]
+    )
+
+    # there should be one token network registered
+    assert len(pathfinding_service.token_networks) == 1
+
+    # Now initialize some channels in this network.
+    # The tuples in channel_descriptions define the following:
+    # (
+    #     p1_index,
+    #     p1_deposit,
+    #     p1_transferred_amount,
+    #     p1_fee,
+    #     p2_index,
+    #     p2_deposit,
+    #     p2_transferred_amount,
+    #     p2_fee
+    # )
+    # Topology:
+    #       /-------------\
+    # 0 -- 1 -- 2 -- 3 -- 4    5 -- 6
+    #  \-------/
+
+    channel_descriptions = [
+        (0, clients[0], 100, 20, 0.0010, 1, clients[1],  50, 10, 0.0015),  # capacities  90 --  60
```

Comment:
I would remove the `clients[x]` here, and create them as local variables below to make the topology easier to read.

Reply:
I think fixture for setting up the channel_descriptions/network would be a better choice.

Reply:
Good point. But we can do that in a different PR.

Reply:
Done.